test/e2e/common/node/lifecycle_hook.go

```diff
@@ -58,6 +59,10 @@ var _ = SIGDescribe("Container Lifecycle Hook", func() {
 			if strings.Contains(targetIP, ":") {
 				targetURL = fmt.Sprintf("[%s]", targetIP)
 			}
+
+			pods, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{})
+			framework.ExpectNoError(err)
+			targetNode = pods.Items[0].Spec.NodeName
```

Comment:
this assume that there is only one pod, also, if the List fails we don't retry, 
I think that we can do the other way around,  pick a random node here with
https://github.com/kubernetes/kubernetes/blob/f7cff077d5693d2c5e199aa18c002a4c25667f4c/test/e2e/framework/node/resource.go#L370-L378

and then schedule all pods on that node